src/ore/src/antichain.rs

```diff
@@ -30,13 +31,19 @@ impl<T: PartialOrder + Ord + Clone> AntichainToken<T> {
         I: IntoIterator<Item = T>,
         F: FnMut(std::vec::Drain<(T, i64)>) + 'static,
     {
-        let mut token = Self {
-            antichain: MutableAntichain::new(),
-            changes: std::rc::Rc::new(std::cell::RefCell::new(ChangeBatch::new())),
-            action: std::rc::Rc::new(std::cell::RefCell::new(action)),
-        };
-        token.advance(values);
-        token
+        let mut antichain = MutableAntichain::new();
+        let changes = std::rc::Rc::new(std::cell::RefCell::new(ChangeBatch::new()));
+        let action = std::rc::Rc::new(std::cell::RefCell::new(action));
+        changes
```

Comment:
maybe worth making this an internal method that both new and advance can call so it doesn't have to be duplicated.